ledger-core/v2/virtual/handlers/vstatereport.go

```diff
@@ -74,5 +105,8 @@ func (s *SMVStateReport) stepProcess(ctx smachine.ExecutionContext) smachine.Sta
 		panic(throw.NotImplemented())
 	}
 
+	ctx.ApplyAdjustment(immutableExecuteSemaphore.NewDelta(immutablePendingCount))
```

Comment:
I thought that this should be done inside SMObject

Reply:
this is SO wrong :) counters that are in VStateReport shouldn't change semaphores, they should change counters.

Reply:
you are right, when I've started the ticket there were no counters so I assumed wrong behavior. Fixed